src/modules/avformat/producer_avformat.c

```diff
@@ -1199,7 +1199,7 @@ static int seek_video( producer_avformat self, mlt_position position,
 			// Remove the cached info relating to the previous position
 			self->current_position = POSITION_INVALID;
 			self->last_position = POSITION_INVALID;
-			av_freep( &self->video_frame );
+			av_frame_unref(self->video_frame);
```

Comment:
If this doesn't turn out to have any unintended side effects, I believe it properly fixes the memory leak.

Reply:
Yes, seems to be good here without side effect. But also, I think we should do
```
--- src/modules/avformat/producer_avformat.c
+++ src/modules/avformat/producer_avformat.c
@@ -3129,3 +3129,5 @@ static void producer_avformat_close( producer_avformat self )
 	av_frame_unref( self->video_frame );
+	av_frame_free( &self->video_frame );
 	av_frame_unref( self->audio_frame );
+	av_frame_free( &self->audio_frame );
 
@@ -3183,3 +3185,3 @@ static void producer_avformat_close( producer_avformat self )
 			av_packet_unref( pkt );
-			free( pkt );
+			av_packet_free( &pkt );
 		}
@@ -3193,3 +3195,3 @@ static void producer_avformat_close( producer_avformat self )
 			av_packet_unref( pkt );
-			free( pkt );
+			av_packet_free( &pkt );
 		}
```